dart/dynamics/BodyNode.cpp

```diff
@@ -888,34 +960,16 @@ void BodyNode::init(Skeleton* _skeleton)
 }
 
 //==============================================================================
-void BodyNode::draw(renderer::RenderInterface* _ri,
-                    const Eigen::Vector4d& _color,
-                    bool _useDefaultColor,
-                    int _depth) const
+void BodyNode::processNewEntity(Entity* _newChildEntity)
 {
-  if (_ri == NULL)
+  if(find(mChildBodyNodes.begin(), mChildBodyNodes.end(), _newChildEntity) !=
+     mChildBodyNodes.end())
```

Comment:
I can see storing Entities and BodyNodes has benefits in the performance. 

What I meant was that the check looks redundant. `BodyNode::processNewEntity(~)` seems to be called by only `addChildBodyNode(~)` and the duplicity check is already done in the function. Is there any possibility that `BodyNode::processNewEntity(~)` can be called in other functions?